lib/rules/no-else-return.js

```diff
@@ -33,7 +35,24 @@ module.exports = {
          * @returns {void}
          */
         function displayReport(node) {
-            context.report({ node, message: "Unnecessary 'else' after 'return'." });
+            context.report({
+                node,
+                message: "Unnecessary 'else' after 'return'.",
+                fix: fixer => {
+                    const sourceCode = context.getSourceCode();
+                    const startToken = sourceCode.getTokenByRangeStart(node.start);
+                    const elseToken = sourceCode.getTokenBefore(startToken);
+                    const source = sourceCode.getText(node);
+                    let fixedSource;
+
+                    if (startToken.type === "Punctuator") {
+                        fixedSource = source.substr(1, source.length - 2);
```

Comment:
Ok, took a longer look at these:

1. Your first example:

```javascript
if (foo) bar()
else return baz
```

Is not an error according to the rule, so this one won't get fixed at all. You can't remove the else in this case, as it changes the meaning of the code.

Your second example is the same. It's not an error according to the rule because the else is not extra, it's necessary to the meaning of the code.

However, if you change both of these examples to replace `bar()` with `return bar()` that would make them error states. In doing that, your assertion about how it gets fixed is incorrect. The else case in my code ([L51](https://github.com/eslint/eslint/pull/7864/files#diff-89a736d61e240631c8907a6aa2c1e4c0R51)) handles it, and it fixes the code correctly.

Your third example is also not applicable unless you change `bar()` to `return bar()`. If that's the case you end up with this:

```javascript
if (foo) { return bar() }
else { return baz } qux()

Where `qux()` is unreachable code.